### PR TITLE
bpo-46775: OSError should call winerror_to_errno unconditionally

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue46775

### CODE REVIEW

File: Objects/exceptions.c
Comment (corona10) [inline Objects/exceptions.c:1637]: see

https://github.com/python/cpython/blob/755be9b1505af591b9f2ee424a6525b6c2b65ce9/PC/errmap.h#L10-L23

**Comment by corona10** [comment]
cc @eryksun

File: Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst
Comment (zooba) [inline Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst]: Since this isn't going in the Windows section, it's worth mentioning that somewhere in the text.

Probably also worth mentioning the user-visible impact, which will be that some (obscure) OSErrors may now be raised as a subclass, and the OSError.errno field may now be different from the winerror field.

Comment (corona10) [reply Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst]: @zooba 
Thanks, Steve, I updated the NEWS.d. 
Please let me know the newly updated NEWS entry is well enough as you want.
If not, you can suggest changing the sentence (SInce I am not a good English writer ;))

File: Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst
Comment (vstinner) [inline Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst:3]: winerror_to_errno() is an internal function, but don't mention it but explains the change to end users. What I understand is that a Windows error is now correctly converted to an ``errno`` number for error numbers greater than or equal to 10000. For example, the socket error WSAEACCES is now converted to EACCES.

Maybe even give the list of errors code with their numbers:

```
     case WSAEINTR: 
     case WSAEBADF: 
     case WSAEACCES: 
     case WSAEFAULT: 
     case WSAEINVAL: 
     case WSAEMFILE: 
```

https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--9000-11999-

Comment (corona10) [reply Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst:3]: Thanks Victor :)

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by corona10** [comment]
I will merge this PR after @zooba 's approval :)
Thanks Victor!

File: Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst
Comment (zooba) [inline Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst]:
```suggestion
the correct errno and may now raise a subclass of :exc:`OSError`.
```

**Review by zooba — APPROVED** [review]
Approved with my fix that I already applied ;)

**Comment by zooba** [comment]
I'll merge once CI is done, unless someone else gets in before me (which is fine).

**Comment by vstinner** [comment]
Oh right, WSAEACCES can now raise a PermissionError for example!

**Pull request merged**
